staging/src/k8s.io/client-go/tools/record/events_cache_test.go

```diff
@@ -278,3 +278,105 @@ func TestEventCorrelator(t *testing.T) {
 		}
 	}
 }
+
+func TestEventSpamFilter(t *testing.T) {
+	spamKeyFuncBasedOnObjectsAndReason := func(e *v1.Event) string {
+		return strings.Join([]string{
+			e.Source.Component,
+			e.Source.Host,
+			e.InvolvedObject.Kind,
+			e.InvolvedObject.Namespace,
+			e.InvolvedObject.Name,
+			string(e.InvolvedObject.UID),
+			e.InvolvedObject.APIVersion,
+			e.Reason,
+		},
+			"")
+	}
+	firstEvent := makeEvent("first", "i am first", makeObjectReference("Pod", "my-pod", "my-ns"))
+	duplicateEvent := makeEvent("duplicate", "me again", makeObjectReference("Pod", "my-pod", "my-ns"))
+	similarEvent := makeEvent("first", "me again", makeObjectReference("Pod", "my-pod", "my-ns"))
+	testCases := map[string]struct {
+		previousEvents  []v1.Event
+		newEvent        v1.Event
+		expectedEvent   v1.Event
+		intervalSeconds int
+		burstSize       int
```

Comment:
ditto for burstSize and expected Skip. Not need to include in test case if they don't differ.

Reply:
burstSize is not included in test case anymore, but expectedSkip actually changes between the test cases :)